cmd/kubeadm/app/preflight/checks.go

```diff
@@ -886,6 +886,11 @@ func RunInitMasterChecks(execer utilsexec.Interface, cfg *kubeadmapi.InitConfigu
 		HTTPProxyCIDRCheck{Proto: "https", CIDR: cfg.Networking.ServiceSubnet},
 		HTTPProxyCIDRCheck{Proto: "https", CIDR: cfg.Networking.PodSubnet},
 	}
+
+	if cfg.ComponentConfigs.Kubelet == nil || cfg.ComponentConfigs.Kubelet.FailSwapOn {
+		checks = append(checks, SwapCheck{})
```

Comment:
it has to happen for linux only.

something else we can do for both init and join is to check if the `fail-swap-on` flag exists in the KubeletExtraArgs in NodeRegistrationOptions for both Init and JoinConfiguration:
https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1#NodeRegistrationOptions

such a flag should override the value of cfg.ComponentConfigs.Kubelet.FailSwapOn.
currently cfg.ComponentConfigs.Kubelet is only available for init.

Reply:
> it has to happen for linux only.

True. will do.

> something else we can do

Does this make sense to do it in this PR or separately?

Reply:
i think it should be in the same PR.

Reply:
> something else we can do for both init and join is to check if the fail-swap-on flag exists in the KubeletExtraArgs 

I'm not sure we should do it as the flag seems to be deprecated:
```
--fail-swap-on  Makes the Kubelet fail to start if swap is enabled on the node. 
(default true) (DEPRECATED: This parameter should be set via the config
file specified by the Kubelet's --config flag.
See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
```

Reply:
as far as we know some of these flags could be a around for "a while" and we depend on them for the kubelet env file.

even if it  leaves us without an option to check if the Join node wants to skip, i think it's OK to to not do it for this reason.

instead i think the joining node should eventually have a custom KubeletConfiguration but that's outside of scope for this PR.

Reply:
> even if it leaves us without an option to check if the Join node wants to skip, i think it's OK to to not do it for this reason

sorry for misunderstanding, so should we do it or not?

If not - please lgtm